# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/ipc/PWindowGlobal.ipdl
# Commit: 812ed5275f0d
# Full Hash: 812ed5275f0db9f92944955f891df2ab68c9f8e7
# Author: Toshihito Kikuchi <tkikuchi@mozilla.com>
# Date: 2021-11-19 04:11:43
# Regressor Bug: 1734099
# File Overlap Count: 5
# Description:
#   Bug 1734099 - Never unload tabs with active RTCPeerConnection instances.  r=peterv,jesup
#   
#   This patch makes sure the Tab Unloading feature does not unload tabs that have
#   active peer connections not to disrupt browsing experience based on WebRTC peer
#   connections.
# ==============================================================================

diff -r a733d049dbda -r 812ed5275f0d dom/ipc/PWindowGlobal.ipdl
--- a/dom/ipc/PWindowGlobal.ipdl	Fri Nov 19 01:03:56 2021 +0200
+++ b/dom/ipc/PWindowGlobal.ipdl	Thu Nov 18 22:59:12 2021 +0000
@@ -190,6 +190,10 @@
   // enum for the valid flags.
   async UpdateBFCacheStatus(uint32_t aOnFlags, uint32_t aOffFlags);
 
+  // Signal whether the first connection is added (aIsAdded = true) or
+  // the last connection is removed (aIsAdded = false).
+  async UpdateActivePeerConnectionStatus(bool aIsAdded);
+
   /**
    * Used to notify the parent when there's a change in the number of requests
    * in the loadgroup. If there are no requests this will be set to Nothing().